test/GPv2AllowListAuthenticator.test.ts

```diff
@@ -26,8 +26,8 @@ describe("GPv2AllowListAuthentication", () => {
 
   describe("addSolver(address)", () => {
     it("should add a solver", async () => {
-      await expect(authenticator.connect(owner).addSolver(solver.address)).to
-        .not.be.reverted;
+      const tx = await authenticator.connect(owner).addSolver(solver.address);
+      await expect(tx.wait()).to.not.be.reverted;
```

Comment:
I think this pattern (which appears in multiple places) doesn't work as intended if the transaction reverts. In this case the first line would panic, so the `expect` in the second line should never fail.
Two alternatives could be:
```
      const tx = authenticator.connect(owner).addSolver(solver.address);
      await expect(tx).to.not.be.reverted;
      await (await tx).wait()
```
or:

```
      const tx = await authenticator.connect(owner).addSolver(solver.address);
      await tx.wait();
```
The first keeps the `expect` but is more verbose. The second is nicer but it's unclear what's being tested.
In all three cases (including the original) we need to remember this pattern and apply it to future unit tests.
To be honest, I like none of them, since they add unnecessary overhead to the logic of the test, but I can't think of a better solution.

Reply:
I agree, but will keep this pattern as it is for now and see how the issue I created is resolved.